Chore/service core improvements

agustingroh opened the pull request:
## **What's Changed**


### Changed
- Migrated from `go-sqlite3` to `modernc.org/sqlite`
### Added
- New structured error handling system with ServiceError type
- ComponentDTO for component data transfer
- Enhanced status determination logic with HTTP status code mapping
- Enhanced error handling with grpc metadata trailers
- Added gRPC GetCountryContributorsByComponents and REST endpoint POST 
- /v2/geoprovenance/countries/components
- Added gRPC GetCountryContributorsByComponent and REST endpoint GET /v2/geoprovenance/countries/component
- Added gRPC GetOriginByComponents and REST endpoint POST /v2/geoprovenance/origin/components
- Added gRPC GetOriginByComponent and REST endpoint GET /v2/geoprovenance/origin/component
- Added environment setup for SQLite on env-setup.sh 
### Fixed
- Improved input validation for PURL requests
- Better error propagation and handling throughout the service
- Enhanced status response messaging with detailed failure information
### Refactored
- Refactored geo provenance service core architecture
- Simplified provenance input conversion logic

agustingroh committed:
chore:SP-2547 Upgrades dependendencies to latest version

agustingroh committed:
chore:2548 Replaces go-sqlite-3 lib by modernc.org/sqlite

agustingroh committed:
chore:SP-2549 Fixes linter issues

agustingroh committed:
chore:SP-2544 Increases unit tests coverage

agustingroh committed:
chore:SP-2558 Updates CHANGELOG.md file

eeisegn committed:
fix makefile pwd case

eeisegn committed:
fix linter issue

eeisegn committed:
add sample default prod config

eeisegn committed:
update ignores

eeisegn committed:
fix default port numbers

eeisegn committed:
update readme filename errors

agustingroh committed:
chore:SP-2566 Fixes linter issues, removes unused code and fixes typos

agustingroh committed:
chore:SP-2567 Adds environment setup for SQLite database

eeisegn committed:
change coverage output to default coverage.txt filename

eeisegn committed:
added and updated config files

eeisegn commented on scripts/env-setup.sh:
```diff
@@ -17,9 +17,14 @@ if [ "$1" = "-h" ] || [ "$1" = "-help" ] ; then
+export LOGS_DIR=/var/log/scanoss/geoprovenance
+export DB_PATH_BASE=/var/lib/scanoss
+export SQLITE_PATH="${DB_PATH_BASE}/db/sqlite/geoprovenance"
+export SQLITE_DB_NAME=base.sqlite
```
Is the geo-provenance going into base also?>

agustingroh replied:
We've never defined it for geo provenance service.  We only added support for sqlite database, but it was never used. Currently, we are using Postgres on production

eeisegn reviewed (commented):
Is the base sqlite file correct for setup?

agustingroh committed:
chore:Sp-2626 Text clean up

agustingroh committed:
chore: refactor geo provenance service

agustingroh committed:
chore:SP-3307 Refactor geo provenance service core architecture

  - Migrate from individual DB connections to shared connection pool
  - Implement dependency injection for use cases and models
  - Add service documentation
  - Enhance error handling and logging throughout the service
  - Update test suite to align with new architecture
  - Improve separation of concerns between service, use case, and model layers

agustingroh committed:
feat:SP-3291 implements new geoprovenance protobuf definition

agustingroh committed:
chore:SP-3309 adds test coverage
                                                                                                                                                       │
  - Add new service tests for country contributors and origin endpoints
  - Fix nil pointer issues in response converters with bounds checking
  - Add extensive unit tests for conversion functions in provenance_support
  - Enhance response handler with additional status determination logic- Adds response_handler_test.go

agustingroh committed:
chore:SP-3312 fixes linter issues and adds golang-ci-lint.yml github workflow

agustingroh committed:
chore:SP-3313 upgrades project dependencies to latest version

agustingroh committed:
chore: updates CHANGELOG.md file
